File: Example/QDNavigationBar/QDConfigEditor/QDColorfulView.m
//
//  QDColorfulView.m
//  QDCustomNavigation_Example
//
//  Created by sinno on 2020/8/6.
//  Copyright © 2020 sinno93. All rights reserved.
//

#import "QDColorfulView.h"
#import <Masonry/Masonry.h>

@interface QDColorfulView ()
@property (nonatomic, strong) UIStackView *stackView;
@end

@implementation QDColorfulView

#pragma mark - Initialization
// 初始化方法
- (instancetype)initWithFrame:(CGRect)frame {
    if (self = [super initWithFrame:frame]) {
        [self configSubviews];
    }
    return self;
}

#pragma mark - Super methods
#pragma mark - Action Methods
#pragma mark - Public Methods

#pragma mark - Private Methods
// 私有的方法
- (void)configSubviews {
    [self addSubview:self.stackView];
    [self.stackView mas_makeConstraints:^(MASConstraintMaker *make) {
        make.edges.equalTo(self).insets(UIEdgeInsetsMake(8, 13, 8, 13));
    }];
    
    NSArray *colorArray = @[UIColor.redColor, UIColor.greenColor, UIColor.blueColor, UIColor.cyanColor, UIColor.magentaColor];
    for (UIColor *color in colorArray) {
        UIView *view = [[UIView alloc] init];
        view.backgroundColor = color;
        [view mas_makeConstraints:^(MASConstraintMaker *make) {
            make.size.mas_equalTo(CGSizeMake(30, 30));
//            make.width.equalTo(view.mas_height);
//            make.height.equalTo(@30);
        }];
        [self.stackView addArrangedSubview:view];
    }
    self.layer.borderColor = UIColor.lightGrayColor.CGColor;
    self.layer.borderWidth = 1.0/UIScreen.mainScreen.scale;
    self.layer.cornerRadius = 15;
}
#pragma mark - Getter && Setter
// Getter和Setter方法
- (UIStackView *)stackView {
    if (!_stackView) {
        UIStackView *view = [UIStackView new];
        view.axis = UILayoutConstraintAxisHorizontal;
        view.alignment = UIStackViewAlignmentCenter;
        view.distribution = UIStackViewDistributionEqualCentering;
        view.spacing = 10.0f;
        
        view.backgroundColor = UIColor.grayColor;
        _stackView = view;
    }
    return _stackView;
}

@end

File: Example/QDNavigationBar/QDConfigEditor/QDColorfulView.swift
//
//  QDColorfulView.swift
//  QDNavigationBar_Example
//
//  Created by zqd on 2024/6/5.
//  Copyright © 2024 CocoaPods. All rights reserved.
//

import UIKit

class QDColorfulView: UIView {
    
    // MARK: - Properties
    
    private var stackView: UIStackView = {
        let stackView = UIStackView()
        stackView.axis = .horizontal
        stackView.alignment = .center
        stackView.distribution = .equalCentering
        stackView.spacing = 10.0
        stackView.backgroundColor = .gray
        return stackView
    }()
    
    // MARK: - Initialization
    
    override init(frame: CGRect) {
        super.init(frame: frame)
        configSubviews()
    }
    
    required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        configSubviews()
    }
    
    // MARK: - Private Methods
    
    private func configSubviews() {
        addSubview(stackView)
        stackView.snp.makeConstraints { make in
            make.edges.equalToSuperview().inset(UIEdgeInsets(top: 8, left: 13, bottom: 8, right: 13))
        }
        
        let colorArray: [UIColor] = [.red, .green, .blue, .cyan, .magenta]
        for color in colorArray {
            let view = UIView()
            view.backgroundColor = color
            stackView.addArrangedSubview(view)
            view.snp.makeConstraints { make in
                make.size.equalTo(CGSize(width: 30, height: 30))
                // Uncomment below lines if you want to make width equal to height
                // make.width.equalTo(view.snp.height)
                // make.height.equalTo(30)
            }
        }
        
        layer.borderColor = UIColor.lightGray.cgColor
        layer.borderWidth = 1.0 / UIScreen.main.scale
        layer.cornerRadius = 15.0
    }
}

